lib/matplotlib/ticker.py

```diff
@@ -1258,19 +1283,17 @@ def format_eng(self, num):
 
         mant = sign * dnum / (10 ** pow10)
 
+        # TODO: shouldn't we raise a warning if self.places < 0?
```

Comment:
I think not mixing two formatting methods would be more legible, e.g.
```
formatted = "{mant:{fmt}}{sep}{prefix}".format(
    mant=mant, prefix=prefix, sep=sep,
    fmt="g" if self.places is None else "d" if self.places == 0 else ".{}f".format(self.places))
```
(possibly moving the last line out, but note that the ability to nest formatting expressions is a real plus for str.format here...)